src/InputFieldDescriptor.php

```diff
@@ -0,0 +1,345 @@
+    {
+        $this->defaultValue = $defaultValue;
+    }
+
+
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @return InputType|(NullableType&Type)
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * @param InputType|(NullableType&Type) $type
+     */
+    public function setType($type): void
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * @return array<string, ParameterInterface>
+     */
+    public function getParameters(): array
+    {
+        return $this->parameters;
+    }
+
+    /**
+     * @param array<string, ParameterInterface> $parameters
+     */
+    public function setParameters(array $parameters): void
+    {
+        $this->parameters = $parameters;
+    }
+
+//    /**
+//     * @return array<string, ParameterInterface>
+//     */
+//    public function getPrefetchParameters(): array
+//    {
+//        return $this->prefetchParameters;
+//    }
+//
+//    /**
+//     * @param array<string, ParameterInterface> $prefetchParameters
+//     */
+//    public function setPrefetchParameters(array $prefetchParameters): void
+//    {
+//        $this->prefetchParameters = $prefetchParameters;
+//    }
+//
+//    public function getPrefetchMethodName(): ?string
+//    {
+//        return $this->prefetchMethodName;
+//    }
+//
+//    public function setPrefetchMethodName(?string $prefetchMethodName): void
+//    {
+//        $this->prefetchMethodName = $prefetchMethodName;
+//    }
+
+    /**
+     * Sets the callable targeting the resolver function if the resolver function is part of a service.
+     * This should not be used in the context of a field middleware.
+     * Use getResolver/setResolver if you want to wrap the resolver in another method.
+     */
+    public function setCallable(callable $callable): void
+    {
+        if ($this->originalResolver !== null) {
+            throw new GraphQLRuntimeException('You cannot modify the callable via setCallable because it was already used. You can still wrap the callable using getResolver/setResolver');
+        }
+        $this->callable = $callable;
+        $this->targetMethodOnSource = null;
+        $this->targetPropertyOnSource = null;
+//        $this->magicProperty = null;
+    }
+
+    public function setTargetMethodOnSource(string $targetMethodOnSource): void
+    {
+        if ($this->originalResolver !== null) {
+            throw new GraphQLRuntimeException('You cannot modify the target method via setTargetMethodOnSource because it was already used. You can still wrap the callable using getResolver/setResolver');
+        }
+        $this->callable = null;
+        $this->targetMethodOnSource = $targetMethodOnSource;
+        $this->targetPropertyOnSource = null;
+//        $this->magicProperty = null;
```

Comment:
Remove unused code

Reply:
the magic property code is to be enabled in a future PR

Reply:
Let's add a comment with any commented out code. In general, we should remove virtually all commented out code, but small lines like this are fine if we can share with others why it's there.